android/player/src/main/java/com/intuit/playerui/android/lifecycle/PlayerViewModel.kt

```diff
@@ -78,7 +80,7 @@ public open class PlayerViewModel(flows: AsyncFlowIterator) : ViewModel(), Andro
             deferredPlayer.getCompleted()
         } else {
             runBlocking {
-                deferredPlayer.await()
+                if (viewModelScope.isActive) deferredPlayer.await() else throw IllegalAccessError("Accessing Player instance when ViewModelScope is no longer active")
```

Comment:
1. Can we do the check outside the `runBlocking` to avoid the perf overhead of creating coroutines and blocking the thread on that coroutine context?

2. If the scope that `deferredPlayer` executes in changes, this check would need to be updated to. It'd probably be better to do `deferredPlayer.isActive` if that's possible to check "activeness" explicitly against the `Deferred` we're trying to access.

Reply:
moved to release so no longer needed to check and return or throw here